tests/integration/pilot/outboundtrafficpolicy/helper.go

```diff
@@ -166,3 +173,38 @@ func RunExternalRequestTest(expected map[string][]string, t *testing.T) {
 			}
 		})
 }
+
+func clusterName(target echo.Instance, port echo.Port) string {
+	cfg := target.Config()
+	return fmt.Sprintf("outbound|%d||%s.%s.%s", port.ServicePort, cfg.Service, cfg.Namespace.Name(), cfg.Domain)
+}
+
+// Wait for the destination to NOT be callable by the client. This allows us to simulate external traffic.
+// This essentially just waits for the Sidecar to be applied, without sleeping.
+func WaitUntilNotCallable(c echo.Instance, dest echo.Instance) error {
```

Comment:
I suspect other tests may need to do something similar.

Perhaps we should rename [CheckOutboundConfig](https://github.com/istio/istio/blob/master/pkg/test/framework/components/echo/common/envoy.go#L107) to `CheckCallable` and add your function as `CheckNotCallable`.

We might also consider adding a `WaitUntilNotCallable` to the echo Instance API.

WDYT?

Reply:
I think this is a pretty niche use case. It should probably be broken out somewhere that can be used by other tests, but I am not sure it is worth making it a first-class API in echo at the same level as wait until callable.

AFAIK the only case this is used is if we have a Sidecar restricting the scope? Are there other cases?

Reply:
I'm not sure it is a nich use case. Testing reachability/non-reachability is pretty central to Istio. The security tests do this sort of thing, for example, but they just sleep for a duration before after applying a policy. It would be better for them to use this method.

Reply:
I thought for the policy stuff the config is pushed, we just block access. If this isn't the case, then absolutely we should bring this into the API. If not, I don't mind bringing it in if you think its a good idea

Reply:
Leaving it here is fine for now.  Let's just get this in ... we can iterate on it.